python_modules/libraries/dagster-dg/dagster_dg/cli/shared_options.py

```diff
@@ -0,0 +1,142 @@
+            help="Specify a directory to use for the cache.",
+        ),
+        click.Option(
+            ["--disable-cache"],
+            is_flag=True,
+            default=DgCliConfig.disable_cache,
+            help="Disable the cache..",
+        ),
+        click.Option(
+            ["--verbose"],
+            is_flag=True,
+            default=DgCliConfig.verbose,
+            help="Enable verbose output for debugging.",
+        ),
+        click.Option(
+            ["--use-component-module", "use_component_modules"],
+            type=str,
+            multiple=True,
+            default=DgCliConfig.__dataclass_fields__["use_component_modules"].default_factory(),
+            hidden=True,
+            help=textwrap.dedent("""
+                Specify a list of remote environment modules expected to contain components.
+                When retrieving the default set of components from the target environment, only
+                components from these modules will be fetched. This overrides the default behavior
+                of fetching all components registered under entry points in the remote environment.
+                This is useful primarily for testing, as it allows targeting of a stable set of test
+                components.
+            """).strip(),
+        ),
+        click.Option(
+            ["--use-dg-managed-environment/--no-use-dg-managed-environment"],
+            is_flag=True,
+            default=DgCliConfig.use_dg_managed_environment,
+            help="Enable management of the virtual environment with uv.",
+        ),
+        click.Option(
+            ["--require-local-venv/--no-require-local-venv"],
+            is_flag=True,
+            default=DgCliConfig.require_local_venv,
+            help="Require use of a local virtual environment (`.venv` found in ancestors of the working directory).",
+        ),
+    ]
+}
+
+# Ensure that these options show up in the help output under the "Global options" group.
+for option in GLOBAL_OPTIONS.values():
+    set_option_help_output_group(option, "Global options")
+
+dg_global_options = make_option_group(GLOBAL_OPTIONS)
+
+# ########################
+# ##### EDITABLE DAGSTER
+# ########################
+
+EDITABLE_DAGSTER_OPTIONS = {
+    not_none(option.name): option
+    for option in [
+        click.Option(
+            ["--use-editable-dagster"],
+            type=str,
+            flag_value="TRUE",
+            is_flag=False,
+            default=None,
+            help=(
+                "Install all Dagster package dependencies from a local Dagster clone. Accepts a path to local Dagster clone root or"
+                " may be set as a flag (no value is passed). If set as a flag,"
+                " the location of the local Dagster clone will be read from the `DAGSTER_GIT_REPO_DIR` environment variable."
+            ),
+        ),
+        click.Option(
+            ["--use-editable-components-only"],
```

Comment:
I'm finding the name here pretty confusing in a way that --use-editable-dagster was not, because it makes me think that this is referring to the components themselves, rather than `dagster-components` the python package.

Maybe this is not the end of the world for a temporary hidden flag that is really only for development, but just flagging. you could consider going even more verbose and make this is like --use-editable-components-package-only

Reply:
yeah I agree it's not clear. Will update to `--use-editable-components-package-only`.